components/suggest/src/db.rs

```diff
@@ -853,22 +859,56 @@ impl<'a> SuggestDao<'a> {
                     )?;
                 }
             }
-            for (index, keyword) in common_details.keywords.iter().enumerate() {
+            let mut last_inserted_full_keyword = ("", 0);
+
+            for keyword in suggestion.keywords() {
+                let full_keyword_id = match (suggestion, keyword.full_keyword) {
+                    // Try to associate full keyword data.  Only do this for AMP, we decided to
+                    // skip it for Wikipedia in https://bugzilla.mozilla.org/show_bug.cgi?id=1876217
+                    (DownloadedAmpWikipediaSuggestion::Amp(_), Some(full_keyword)) => {
+                        Some(if full_keyword == last_inserted_full_keyword.0 {
+                            last_inserted_full_keyword.1
+                        } else {
+                            self.conn.execute(
+                                "INSERT INTO full_keywords(
+                                    suggestion_id,
+                                    full_keyword
+                                 )
+                                 VALUES(
+                                    :suggestion_id,
+                                    :keyword
+                                 )",
+                                named_params! {
+                                    ":keyword": full_keyword,
+                                    ":suggestion_id": suggestion_id,
+                                },
+                            )?;
+                            let full_keyword_id = self.conn.last_insert_rowid();
```

Comment:
We often use `RETURNING id` in the `Insert` SQL to get the last insert rowid, can we do that here as well?

Reply:
TIL.  I updated the code to use that.